posthog/api/decide.py

```diff
@@ -222,9 +222,13 @@ def get_decide(request: HttpRequest):
                 else False
             )
 
-            if settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS and str(team.id) in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS:
-                if random() < settings.NEW_ANALYTICS_CAPTURE_SAMPLING_RATE:
-                    response["analytics"] = {"endpoint": settings.NEW_ANALYTICS_CAPTURE_ENDPOINT}
+            if str(team.id) not in settings.NEW_ANALYTICS_CAPTURE_EXCLUDED_TEAM_IDS:
+                if (
+                    "*" in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS
+                    or str(team.id) in settings.NEW_ANALYTICS_CAPTURE_TEAM_IDS
+                ):
+                    if random() < settings.NEW_ANALYTICS_CAPTURE_SAMPLING_RATE:
```

Comment:
observation: `random` has a range `[0.0, 1.0)`, so if we ever set `NEW_ANALYTICS_CAPTURE_SAMPLING_RATE = 0` some executions can get to this block.

It's very unlikely, but if we ever need to disable this, it's good to keep in mind we should set `NEW_ANALYTICS_CAPTURE_TEAM_IDS` to an empty set instead of bringing `NEW_ANALYTICS_CAPTURE_SAMPLING_RATE` down to 0.